tests/unit/test_parts.py

```diff
@@ -64,6 +66,88 @@ def test_parts_lifecycle_run(new_dir):
     assert Path(lifecycle.prime_dir, "foo.txt").is_file()
 
 
+@pytest.mark.parametrize("step_name", ["pull", "overlay", "build", "stage", "prime"])
```

Comment:
something like this avoids the if block below
```suggestion
@pytest.mark.parametrize("step_name,last_step", [("pull", None), ("overlay",steps.PULL), ("build", steps.OVERLAY), ("stage", steps.BUILD), ("prime", steps.STAGE)])
```

Reply:
Much better, will also change this in the original test in Snapcraft.